src/librustc_mir/interpret/operand.rs

```diff
@@ -314,8 +314,9 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
 
     // Turn the wide MPlace into a string (must already be dereferenced!)
     pub fn read_str(&self, mplace: MPlaceTy<'tcx, M::PointerTag>) -> InterpResult<'tcx, &str> {
-        let len = mplace.len(self)?;
-        let bytes = self.memory.read_bytes(mplace.ptr, Size::from_bytes(len))?;
+        let bytes_mplace = self.mplace_field(mplace, 0)?;
```

Comment:
potential site for the perf regression. We should probably check whether we can make `mplace_field` a no-op in case of `repr(Rust)` with one field or `repr(transparent)` types.

I think the main problem here is that we compute the field's layout even for cases like this one where we don't actually need the layout. We've seen major regressions in miri due to a single additional call to `tcx.layout_of` before.